dap_aria_mapping/pipeline/semantic_taxonomy/make_taxonomy.py

```diff
@@ -155,16 +167,25 @@ def plot_imbalanced(
         cluster_configs, embeddings, embeddings_2d=embeddings_2d, imbalanced=imbalanced
     )
 
-    if all([args.production, args.test is False]):
+    if all([args.production, args.test is False, args.sample_frac is None]):
         logger.info("Saving clustering results to S3")
         s3 = boto3.client("s3")
         s3.put_object(
             Body=pickle.dumps(cluster_outputs),
             Bucket=BUCKET_NAME,
             Key=f"{OUTPUT_DIR}/raw_outputs/semantic_{args.cluster_method}.pkl",
         )
+    
+    if all([args.production, args.test is False, args.sample_frac is not None]):
+        logger.info("Saving clustering results to S3")
+        s3 = boto3.client("s3")
+        s3.put_object(
+            Body=pickle.dumps(cluster_outputs),
+            Bucket=BUCKET_NAME,
+            Key=f"{OUTPUT_DIR}/raw_outputs/semantic_{args.cluster_method}_{str(int(args.sample_frac*100))}.pkl",
+        )
 
-    if args.plot:
+    if all(args.plot, args.sample_frac is None):
```

Comment:
This will not work, all takes exactly one argument and these are two. Do `all([args.plot, args.sample_frac is None]):` insstead.